ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Extended.hs

```diff
@@ -53,21 +53,28 @@ applyExtLedgerState :: ( UpdateLedger blk
                        , ProtocolLedgerView blk
                        , HasCallStack
                        )
-                    => NodeConfig (BlockProtocol blk)
+                    => Bool -- ^ Was this block previously applied to the given ledger state?
+                    -> NodeConfig (BlockProtocol blk)
                     -> blk
                     -> ExtLedgerState blk
                     -> Except (ExtValidationError blk) (ExtLedgerState blk)
-applyExtLedgerState cfg blk ExtLedgerState{..} = do
+applyExtLedgerState prevApplied cfg blk ExtLedgerState{..} = do
     ledgerState'         <- withExcept ExtValidationErrorLedger $
                               applyChainTick
                                 (ledgerConfigView cfg)
                                 (blockSlot blk)
                                 ledgerState
-    ledgerState''        <- withExcept ExtValidationErrorLedger $
+    ledgerState''        <- if not prevApplied
```

Comment:
Any reason we use a `not` here, rather than flipping the `then` and `else` branches?

Reply:
This is the kind of thing that's easy to read wrong, or get wrong, hence suggestion to use a new type rather than `Bool`.

Reply:
No, no particular reason for this. I'll switch over to a `newtype`